includes/admin/tools/class-settings-import.php

```diff
@@ -0,0 +1,681 @@
+						Give_Settings_Import::import_success();
+					}
+
+					if ( self::check_for_dropdown_or_import() == false ) {
+						$step = Give_Settings_Import::get_step();
+						?>
+                        <tr valign="top">
+                            <th></th>
+                            <th>
+                                <input type="submit"
+                                       class="button button-primary button-large button-secondary <?php echo 'step-' . $step; ?>"
+                                       id="recount-stats-submit"
+                                       value="<?php esc_attr_e( 'Submit', 'give' ); ?>"/>
+                            </th>
+                        </tr>
+						<?php
+					}
+					?>
+                    </tbody>
+                </table>
+            </section>
+			<?php
+		}
+
+		static function import_success() {
+			$report  = give_import_donation_report();
+			$total   = (int) $_GET['total'];
+			$total   = $total - 1;
+			$success = (bool) $_GET['success'];
+			?>
+            <tr valign="top" class="give-import-dropdown">
+                <th colspan="2">
+                    <h2>
+						<?php
+						if ( $success ) {
+							echo sprintf( __( 'Import complete! %s donations imported', 'give' ), '<strong>' . $total . '</strong>' );
+						} else {
+							echo sprintf( __( 'Failed to import %s donations', 'give' ), '<strong>' . $total . '</strong>' );
+						}
+						?>
+                    </h2>
+
+					<?php
+					$text      = __( 'Import Donation', 'give' );
+					$query_arg = array(
+						'post_type' => 'give_forms',
+						'page'      => 'give-tools',
+						'tab'       => 'import',
+					);
+					if ( $success ) {
+						$query_arg = array(
+							'post_type' => 'give_forms',
+							'page'      => 'give-payment-history',
+						);
+						$text      = __( 'View Donations', 'give' );
+					}
+					?>
+					<?php
+					if ( ! empty( $report['duplicate_donor'] ) ) {
+						?>
+                        <p>
+							<?php echo esc_html( wp_sprintf( '%s duplicate donors detected', $report['duplicate_donor'] ), 'give' ); ?>
+                        </p>
+						<?php
+					}
+					?>
+
+					<?php
+					if ( ! empty( $report['create_donor'] ) ) {
+						?>
+                        <p>
+							<?php echo esc_html( wp_sprintf( '%s donors created', $report['create_donor'] ), 'give' ); ?>
+                        </p>
+						<?php
+					}
+					?>
+
+					<?php
+					if ( ! empty( $report['create_form'] ) ) {
+						?>
+                        <p>
+							<?php echo esc_html( wp_sprintf( '%s donations forms created', $report['create_form'] ), 'give' ); ?>
+                        </p>
+						<?php
+					}
+					?>
+
+					<?php
+					if ( ! empty( $report['duplicate_donation'] ) ) {
+						?>
+                        <p>
+							<?php echo esc_html( wp_sprintf( '%s duplicate donations detected', $report['duplicate_donation'] ), 'give' ); ?>
+                        </p>
+						<?php
+					}
+					?>
```

Comment:
@raftaar1191 Collect import report in an array and then print them once

Reply:
Done